graph/src/log/mod.rs

```diff
@@ -0,0 +1,31 @@
+use std::fmt::{Display, Error, Formatter};
+
+pub enum LogCode {
+    SubgraphStartFailure,
+    SubgraphExecutionFailure,
+    SubgraphExecutionFailureNotRecorded,
+    BlockSyncStatus,
+}
+
+impl Display for LogCode {
+    fn fmt(&self, f: &mut Formatter) -> Result<(), Error> {
+        let value = match self {
+            LogCode::SubgraphStartFailure => "SubgraphStartFailure",
+            LogCode::SubgraphExecutionFailure => "SubgraphExecutionFailure",
```

Comment:
I'm not a fan of introducing the term `execution` here. We already use `indexing`, `syncing` and `processing` in different places. Let's make `SubgraphSyncing...`.